### PR TITLE
Fix endian issue in jemalloc_bins for s390x

### PR DESCRIPTION
**Pull request by HarryLeeIBM**
On s390x, getJeMallocValue() at  StorageSystemJemalloc.cpp:28 returns wrong value when querying value of "arenas.nbins" because of endian issue. This causes the failure of functional test `01161_all_system_tables`.

For example, the function is supposed to return `0x24` but it returns `0x2400000000` instead.

The reason is that mallctl() returns an UInt32 value for "arenas.nbins" (the value of `size` is set to 4 after the call), but the UInt64 `value` interprets it as `0x2400000000` (taking `0x24` for an example) on s390x.

The fix is to right shift 32 bit of the UInt64 value for s390x when the returned size is 4 in order to get correct result for s390x.

### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed endian issue in jemalloc_bins system table for s390x.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Storages/System/StorageSystemJemalloc.cpp [hunk 1]
```diff
@@ -25,6 +25,12 @@ UInt64 getJeMallocValue(const char * name)
     UInt64 value{};
     size_t size = sizeof(value);
     mallctl(name, &value, &size, nullptr, 0);
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
```
Comment (alexey-milovidov) [inline src/Storages/System/StorageSystemJemalloc.cpp:28]: Missing comments.

File: src/Storages/System/StorageSystemJemalloc.cpp [hunk 2]
```diff
@@ -25,6 +25,12 @@ UInt64 getJeMallocValue(const char * name)
...
+    if (size == 4)
+    {
+        value >>= 32;
```
Comment (alexey-milovidov) [inline src/Storages/System/StorageSystemJemalloc.cpp:31]: Why don't fix jemalloc instead?

File: src/Storages/System/StorageSystemJemalloc.cpp (hunk 1 above)
Comment (HarryLeeIBM) [reply src/Storages/System/StorageSystemJemalloc.cpp:31]: Comments were added.

File: src/Storages/System/StorageSystemJemalloc.cpp (hunk 2 above)
Comment (HarryLeeIBM) [reply src/Storages/System/StorageSystemJemalloc.cpp:34]: @alexey-milovidov, jemalloc does the correct thing: for some queries like "arenas.nbins", mallctl() fills the output with 4 bytes of integer value and ALSO set the "size" to 4.
The problem is that the code was using an UInt64 variable for an 32 bit output. On little-endian machines it is correct, but on big-endian machine the value would be 2^32 times larger than the actual result.
The fix checks if the actual output size (which is set by mallctl()) is 4. If it is, we need to right-shift the UInt64 integer by 32 on big-endian machines. (Another way to fix it is to use UInt32 for output for those 32 bit values according to names, but the current fix is much more elegant).

**Review by alexey-milovidov — APPROVED** [review]

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit 26b17f88d5e98df38f67c96c69d60824eccecf33 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/54517/26b17f88d5e98df38f67c96c69d60824eccecf33/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by HarryLeeIBM** [comment]
@alexey-milovidov, the CI test for the PR seems OK, could we get this PR merged? We'd really like to have this fix in the September release, thanks!

**Comment by HarryLeeIBM** [comment]
@alexey-milovidov, I triggered the build again, this time there is no errors any more. Thanks!

**Comment by Felixoid** [comment]
Merged since no changes were made after the review

**Pull request merged**